curp/src/server/raw_curp/tests.rs

```diff
@@ -521,6 +525,31 @@ fn recover_from_spec_pools_will_pick_the_correct_cmds() {
     });
 }
 
+/*************** tests for leader retires **************/
+
+/// To ensure #331 is fixed
+#[traced_test]
+#[test]
+fn leader_retires_after_log_compact_will_succeed() {
+    let curp = RawCurp::new_test(
+        3,
+        MockCEEventTxApi::<TestCommand>::default(),
+        mock_role_change(),
+    );
+    let mut log_w = curp.log.write();
+    for _ in 1..=20 {
+        let cmd = Arc::new(TestCommand::default());
+        log_w.push_cmd(0, cmd).unwrap();
+    }
+    log_w.last_as = 20;
+    log_w.last_exe = 20;
+    log_w.commit_index = 20;
+    log_w.compact();
+    drop(log_w);
+
+    curp.leader_retires();
+}
```

Comment:
Add some code for test `leader_retires` clear sp and ucp? because under the current logic, `leader_retires` clear nothing.

Reply:
added